parts/linux/cloud-init/artifacts/cse_main.sh

```diff
@@ -354,6 +354,26 @@ else
             
         fi
         aptmarkWALinuxAgent unhold &
+    elif [[ $OS == $MARINER_OS_NAME ]]; then
+        if [ "${ENABLE_UNATTENDED_UPGRADES}" == "true" ]; then
+            if [ "${IS_KATA}" == "true" ]; then
+                # Currently kata packages must be updated as a unit (including the kernel which requires a reboot). This can
+                # only be done reliably via image updates as of now so never enable automatic updates.
+                echo 'EnableUnattendedUpgrade is not supported by kata images, will not be enabled'
+            else
+                # By default the dnf-automatic is service is notify only in Mariner.
+                # Enable the automatic install timer and the check-restart timer.
+                # Stop the notify only dnf timer since we've enabled the auto install one.
+                # systemctlDisableAndStop adds .service to the end which doesn't work on timers.
+                systemctl disable dnf-automatic-notifyonly.timer
+                systemctl stop dnf-automatic-notifyonly.timer
+                # At 6:00:00 UTC (1 hour random fuzz) download and install package updates.
+                systemctl unmask dnf-automatic-install.service || exit $ERR_SYSTEMCTL_START_FAIL
```

Comment:
You `systemctl disable dnf-automatic-install.service` in the initial VHD setup, just confirming if you don't need to `systemctl enable dnf-automatic-install.service` here to undo that, or is the timer still able to start the service once its unmasked.

Reply:
I think this works as long as you start the timer, I had a similar issue for ubuntu

Reply:
pasting from a bit above:
```
      systemctl unmask apt-daily.service apt-daily-upgrade.service
      systemctl enable apt-daily.service apt-daily-upgrade.service
      systemctl enable apt-daily.timer apt-daily-upgrade.timer
```

Reply:
My understanding is that a timer triggered service is not actually started, but only triggered by the timer